diff --git a/bootable/recovery/crypto/vold_decrypt/vold_decrypt.cpp b/bootable/recovery/crypto/vold_decrypt/vold_decrypt.cpp
--- a/bootable/recovery/crypto/vold_decrypt/vold_decrypt.cpp
+++ b/bootable/recovery/crypto/vold_decrypt/vold_decrypt.cpp
@@ -242,13 +242,15 @@
 bool Service_Exists(const string& initrc_svc) {
 	return (Get_Service_State(initrc_svc) != "error");
 }
 
+#ifdef TW_CRYPTO_SYSTEM_VOLD_SERVICES
 bool Is_Service_Running(const string& initrc_svc) {
 	return (Get_Service_State(initrc_svc) == "running");
 }
 
 bool Is_Service_Stopped(const string& initrc_svc) {
 	return (Get_Service_State(initrc_svc) == "stopped");
 }
+#endif
 
 bool Start_Service(const string& initrc_svc, int utimeout = SLEEP_MAX_USEC) {
@@ -296,5 +298,7 @@
+#ifdef TW_CRYPTO_SYSTEM_VOLD_SERVICES
 bool will_VendorBin_Be_Symlinked(void) {
 	return (!is_Vendor_Mounted() && TWFunc::Path_Exists("/system/vendor"));
 }
+#endif
 
 bool Symlink_Vendor_Folder(void) {
